dsc/workflows/base/__init__.py

```diff
@@ -21,51 +28,101 @@
+                submissions will be uploaded.
             batch_id: Unique identifier for a 'batch' deposit that corresponds
                 to the name of a subfolder in the workflow directory of the S3 bucket.
                 This subfolder is where the S3 client will search for bitstream
                 and metadata files.
-            collection_handle: The handle of the DSpace collection to which
-                submissions will be uploaded.
-            output_queue: The SQS output queue used for retrieving result messages
-                from the workflow's submissions.
         """
-        self.workflow_name: str = workflow_name
-        self.submission_system: str = submission_system
-        self.email_recipients: list[str] = email_recipients
-        self.metadata_mapping: dict = metadata_mapping
-        self.s3_bucket: str = s3_bucket
-        self.batch_id: str = batch_id
-        self.collection_handle: str = collection_handle
-        self.output_queue: str = output_queue
+        self.batch_id = batch_id
+        self.collection_handle = collection_handle
 
     @property
     def batch_path(self) -> str:
         return f"{self.workflow_name}/{self.batch_id}"
 
+    @property
+    def metadata_mapping(self) -> dict:
+        with open(self.metadata_mapping_path) as mapping_file:
+            return json.load(mapping_file)
+
+    @final
+    @classmethod
+    def load(
+        cls, workflow_name: str, collection_handle: str, batch_id: str
+    ) -> BaseWorkflow:
+        """Return configured workflow class instance.
+
+        Args:
+            workflow_name: The label of the workflow. Must match a key from
+            config.WORKFLOWS.
+            collection_handle: The handle of the DSpace collection to which the batch will
+            be submitted.
+            batch_id: The S3 prefix for the batch of DSpace submissions.
+        """
+        workflow_class = cls.get_workflow(workflow_name)
+        return workflow_class(
+            collection_handle=collection_handle,
+            batch_id=batch_id,
+        )
+
+    @final
+    @classmethod
+    def get_workflow(cls, workflow_name: str) -> type[BaseWorkflow]:
+        """Return workflow class.
+
+        Args:
+            workflow_name: The label of the workflow. Must match a workflow_name attribute
+            from BaseWorkflow subclass.
+        """
+        for workflow_class in BaseWorkflow.__subclasses__():
+            if workflow_name == workflow_class.workflow_name:
+                return workflow_class
+        raise InvalidWorkflowNameError(f"Invalid workflow name: {workflow_name} ")
+
+    def reconcile_bitstreams_and_metadata(self) -> tuple[set[str], set[str]]:
+        """Reconcile bitstreams against metadata.
+
+        Generate a list of bitstreams without item identifiers and item identifiers
+        without bitstreams. Any discrepancies will be addressed by the engineer and
+        stakeholders as necessary.
```

Comment:
I really like this docstring.  I think too often we (at least I) get wrapped up in docstrings being about a technical explanation of technical code, I think this does a fantastic job of situating what this code _does_ in the context when when it's called and by whom.  Nice work, that's not an easy task.

Reply:
Thanks, I'm getting more sensitive to that on this app as I realize the "why" for some of this is not self-evident if you haven't been doing this work in the past